### DISCUSSION TITLE
Adding power stage to the ECU

### DISCUSSION THREAD

**Discussion by DG-PROPULSION** [Jehanzeb1973/Ardu_ECU#11] [Ideas]
Hi @Jehanzeb1973 
I know this seems counterintuitive to the idea of this project BUT what do you think about adding a power stage to the ECU for the fuel pump and starter motor?
Do you have a schematic that we can add to the ECU schematic and then add that to ECU PCB as an option?

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
Traditionally Micro Jet engines had Brushed Starter and Brushed fuel pumps. That is very easy to integrate in Ardu ECU module electronically . My first thought would be to add two additional LR7843 modules. One will control Starter motor and the second will control fuel pump. This needs to be tested but I think controlling DC motor speed with LR7843 will be fairly simple.
Newer jet engines use brushless motor for starter and fuel pump as they are small, light and more powerfull e.g. Swiwin engines. For these types of engines, the design will be complex and basically you need to build two brushless ESC into the PCB. 
Regards

Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/




From: dstromeng
Sent: Tuesday, 12 September 2023 6:57 am
To: Jehanzeb1973/Ardu_ECU
Cc: Jehanzeb1973; Mention
Subject: [Jehanzeb1973/Ardu_ECU] Adding power stage to the ECU (Discussion#11)

Hi @Jehanzeb1973
I know this seems counterintuitive to the idea of this project BUT what do you think about adding a power stage to the ECU for the fuel pump and starter motor?
Do you have a schematic that we can add to the ECU schematic and then add that to ECU PCB as an option?

**Reply by DG-PROPULSION** [reply]
Hi @Jehanzeb1973 
Yea I am thinking if we can make 2 x BLDC ESCs into the schematic.
Is it doable and do you have any ESC schematics.

**Comment by Jehanzeb1973** [comment]
Hi @dstromeng
There are many open source ESC’s schematics available on internet you can use any of these. E.g. bluerobotics/BlueESC: Simple, open-source ESC that uses the SimonK firmware. (github.com)
Open source ESC Arduino speed controller (electronoobs.com)

Open Source Electronic Speed Controller ‘VESC’ Brushless Motor Design by Benjamin Vedder « Adafruit Industries – Makers, hackers, artists, designers and engineers!

I have not tested any of these so cant say which one is better but this  is the route I will take to add ESC in Ardu ECU
Regards


Jehanzeb Khan
DIGIPAK Pvt. Limited
Mob: +92(0333) 5277770


http://DigiPak.org/

**Comment by EFlynn7778** [comment]
The existing LR7843 Modules are overkill for the solenoids. They could be used for brushed starter and pump and additional FET's added that are more suitable for the valves.  It would be highly ideal to have functionality for brushed engines.  These are the engines that have aging ECU's and people are looking for replacements.  Brushless engines most often will be purchased with an ECU and this project being brushless only will get little attention other than from those who want to pursue it merely for academic purposes.  It would be fairly trivial to add functionality for both with it being selectable.   Also, integrating the brushless controls on the main ECU board is not a good idea. They are a relatively high failure rate device and it would be much less costly to replace it as a module.
